lib/libzfs/libzfs_sendrecv.c

```diff
@@ -2725,10 +2725,137 @@ recv_incremental_replication(libzfs_handle_t *hdl, const char *tofs,
 	return (needagain);
 }
 
+/*
+ * Update stream_avl based on override properties:
+ * 1. remove properties from "props" e "snapprops" nvlists if they are present
+ * as boolean values in override_props
+ * 2. add or overwrite string properties from override_props only on the
+ * ancestor "props" and remove them from child "props" and "snapprops" so they
+ * are (possibly) inherited.
+ */
+static int
+props_override(libzfs_handle_t *hdl, avl_tree_t *stream_avl, nvlist_t *props,
+    recvflags_t *flags, const char *errbuf)
+{
+	nvlist_t *noprops;
```

Comment:
**N**ew **O**verride **Props** ... override_props would definitely be better but i started using abbreviated variable names because keeping all these nested loop (to deal with nested nvlist) in 80 cols was becoming a real challange ...